testgrid/config/config.proto

```diff
@@ -52,6 +71,9 @@ message TestGroup {
   // deprecated - always set to true
   bool is_external = 25;
 
+  // Specifies the test name for a test.
+  TestNameConfig test_name_config = 26;
```

Comment:
I am using it for ci-kubernetes-node-kubelet-1.4 job to start with. If it works it can be a default option.